cookin/app/views/index_view.py

```diff
@@ -1,6 +1,58 @@
 from django.shortcuts import render
 from app.forms import SearchForm
 from app.models import Recipe
+from app.models import Rating
+from django.utils.text import slugify
+
+def average_cost(recipe):
+    cost = recipe.estimated_cost
+    qs = Rating.objects.filter(rating_recipe__exact=recipe.id)
+    recipe_ratings = qs.all()
+    if len(recipe_ratings) == 0:
+        return cost
+
+    cost_sum = 0
+    num_estimates = 0
+    if cost:
+        cost_sum += cost
+        num_estimates += 1
+    for rating in recipe_ratings:
+        if rating.rating_price:
+            cost_sum += rating.rating_price * 1.0
+            num_estimates += 1
+    return cost_sum / num_estimates
+
+
+def filter_recipes(recipes, max_time, dietary_restrictions, max_cost, exclude_tools):
+    if not max_time:
```

Comment:
These probably could be done by setting default values for function parameters, e.g. def filter_recipes(recipes, exclude_tools=[])